Directory.Build.props

```diff
@@ -192,6 +192,8 @@
   <!-- The location of the local installation of the .NET Core shared framework. -->
   <PropertyGroup>
     <LocalDotNetRoot>$(RepoRoot).dotnet\</LocalDotNetRoot>
+    <LocalDotNetRoot Condition="'$(DotNetBuild)' == 'true'">DontDareMutateThisSDK</LocalDotNetRoot>
```

Comment:
This is a big area that needs resolution. A number of projects require the local SDK to find the just built ref pack. The VMR does not have a local SDK that can be mutated. Aspnetcore should switch to an approach that doesn't require mutation.

Reply:
For reference, in dotnet/runtime, we set `GenerateErrorForMissingTargetingPacks` to `false` and manually add a `ResolvedTargetingPack` item in an MSBuild target that runs after `ResolveFrameworkReferences` that points to our built targeting pack in our artifacts directory.

See [this file](https://github.com/dotnet/runtime/blob/main/eng/targetingpacks.targets) for examples (and the `UpdateLocalTargetingAndRuntimePack` target in particular).

Reply:
What exactly does `DotNetBuild` cover?

Reply:
Any build that is the ".Net product build". So that would be the repo built in isolation as if it was built in the VMR, the inner repo build, the build if it's run in VMR, etc. It's: "If in Inner Build, Outer Build, or Orchestrated Build"